docs/source/cuda/cuda_array_interface.rst

```diff
@@ -69,13 +69,213 @@ The following are optional entries:
 
   If ``None`` then all values in **data** are valid. All elements of the mask
   array should be interpreted only as true or not true indicating which
-  elements of this array are valid. This has the same definition as ``mask``
-  in the `numpy array interface`_.
+  elements of this array are valid. This has the same definition as *mask* in
+  the `numpy array interface`_.
 
   .. note:: Numba does not currently support working with masked CUDA arrays
-            and will raise a ``NotImplementedError`` exception if one is passed
+            and will raise a `NotImplementedError` exception if one is passed
             to a GPU function.
 
+- **stream**: ``integer``
+
+  An optional stream upon which synchronization may take place. See the
+  :ref:`cuda-array-interface-synchronization` section below.
+
+- **sync**: ``bool``
+
+  Generally, if **sync** is given and is ``True``, then the consumer must either
+  synchronize on the provided **stream** at the point of consumption, or enqueue
+  operations on the data on the given **stream**. See
+  :ref:`cuda-array-interface-synchronization` for further details.
```

Comment:
Why do we need both?  Can we say that if **stream** is present then the consumer must either
  synchronize on the provided **stream** at the point of consumption, or enqueue
  operations on the data on the given **stream**?

Reply:
+1 to @sanjoy's comment that if a stream is provided it is implied that `sync=True` and that the consumer has the choice to synchronize or continue using that stream

Reply:
I think it makes sense to combine them. 

If we don't, then we at least need to specify that they must come in pair (because they're listed as "optional"). In other words, when `sync` is defined in CAI, so is `stream` (it can be `None`, but the key must exist), and vice versa. This would save us a lot of headache to check for corner cases...

Reply:
`stream` and `sync` are now combined into just `stream`, and if `stream` is present then the consumer must sync on the provided stream at the point of consumption, or enqueue operations on that stream.